### PR TITLE
Add image caption offscreen text

### LINKED ISSUE #464: Appending caption text for screen readers
**Is your feature request related to a problem? Please describe.**
Without appending text that denotes a caption, a screen reader user won't know that text being read out is the caption of an image. 

**Describe the solution you'd like**
We should append `Image caption, ` to the `figcaption`. This text should be visually hidden.

**Additional Context**
Similar text will also be appended to a Video, which this is out of scope for this issue.

### PR DESCRIPTION
**Pull request by pjlee11**
_Adds offscreen text to prepend the image caption, required for screen readers. As part of this PR I've split out the VisuallyHiddenText component into it's own component as it's now reused in multiple places_

* [x] Fixes #464 
* ~[ ] Tests added for new features~

* [x] Test engineer approval

**Testing instructions**

Using a variety of screen readers ensure the image caption has offscreen text which reads `Image caption,`

Screenshot of offscreen text 
image

### CODE REVIEW

**Comment by pjlee11** [comment]
blocked by https://github.com/bbc/simorgh/issues/482

File: src/app/components/Figure/index.jsx
Comment (sareh) [inline src/app/components/Figure/index.jsx]: Could you please store the text in a constant - this will make it easier to localise this component later. Also, you can then add a space after the `,` without the linter removing it automatically. :)

File: src/app/components/Figure/Caption/index.stories.jsx
Comment (sareh) [inline src/app/components/Figure/Caption/index.stories.jsx]: I'd suggest having two stories here. One which is without the visually hidden text and one without it. This will highlight the different ways of using the component.

File: src/app/components/Figure/Caption/index.test.jsx
Comment (sareh) [inline src/app/components/Figure/Caption/index.test.jsx]: Are you sure you meant to make this change? Shouldn't the text be passed in as a child, rather than as a prop in this component?

**Review by sareh — CHANGES_REQUESTED** [review]
Looks good. I've just added a few comments.

File: src/app/components/Figure/Caption/index.stories.jsx
Comment (pjlee11) [reply src/app/components/Figure/Caption/index.stories.jsx]: I'd disagree with this as the `VisuallyHiddenText` is an accessibility requirement so making it look optional in the storybook examples is misleading for people who reuse this component in the future. Due to the fact that Figure is actually what passes in the visually hidden text it's currently difficult to ensure the offscreen text is passed into the caption. However in the future when translations are being passed into component we would be able to ensure an `offscreenCaptionText` prop is passed in and then passed to the Caption component. Currently we will have to hardcode this

File: src/app/components/Figure/Caption/index.test.jsx
Comment (pjlee11) [reply src/app/components/Figure/Caption/index.test.jsx]: I think this got lost in merge conflicts, will address

File: src/app/components/Figure/index.jsx
Comment (pjlee11) [reply src/app/components/Figure/index.jsx]: Done 😎

File: src/app/components/Figure/Caption/index.stories.jsx
Comment (sareh) [reply src/app/components/Figure/Caption/index.stories.jsx]: @pjlee11 If you want the visually hidden text to be an immovable part of the Caption, you should not include it here in the Figure in this way. It should be included in the Caption, in here (https://github.com/bbc/simorgh/blob/latest/src/app/components/Figure/Caption/index.jsx).  

Then within `Figure`, you would make no changes at all. 

Thinking through this point, I'd recommend taking this approach, since the `VisuallyHiddenText` shouldn't be passed in as a child element of the Figure.

Comment (pjlee11) [reply src/app/components/Figure/Caption/index.stories.jsx]: This is now addressed - thanks for pointing out React `children` props

**Review by radiocontrolled — COMMENTED** [review]
Looks good to me. I've tested this on an iPhone 5s. I like the  compatibility of this solution and I would like to see our reasoning for choices like this placed in docs/ in due course. 

I previously wrote: "There's a failing Cypress test - Article Meta test - should have script to fetch bundle - that's present on latest." Ignore that, I was running the wrong script, not `npm run test:e2e`.

**Review by radiocontrolled — DISMISSED** [review]

File: src/app/components/VisuallyHiddenText/index.jsx
Comment (sareh) [inline src/app/components/VisuallyHiddenText/index.jsx:2]: This component should also have a test file as well as a stories file. 

The reason for the stories file despite the text not being visible within storybook, is that the existence of the component would be made clear and also to enable easier accessibility testing of this component in isolation.

**Review by sareh — CHANGES_REQUESTED** [review]
Looks great. Just added a comment below.

**Review by sareh — DISMISSED** [review]
👍 Thanks for the additions. After this has been code reviewed, this ticket should be tested with screenreaders. Instructions are here: https://bbc-news.github.io/accessibility-news-and-you/accessibility-and-supported-assistive-technology

You can use this url: https://simorghstorybook.now.sh/iframe?selectedKind=Figure&selectedStory=with%20a%20caption 

Unfortunately, this url does not exist in the localhost version of Storybook, so this testing will need to happen after merging. If there are problems, then we can spin up a new issue to address them. 
@pjlee11 @jamesbrumpton

**Review by radiocontrolled — DISMISSED** [review]

**Review by sareh — DISMISSED** [review]

File: src/app/components/Figure/Caption/index.jsx
Comment (sareh) [inline src/app/components/Figure/Caption/index.jsx]: Looking at React proptypes docs again, that's reminded me you can change this to be `node.isRequired`. https://reactjs.org/docs/typechecking-with-proptypes.html

**Review by sareh — APPROVED** [review]

**Pull request merged**
